README.rst
Google Cloud Python HappyBase Client
====================================

    Python idiomatic client for `Google Cloud Platform`_ HappyBase emulation.

.. _Google Cloud Platform: https://cloud.google.com/

|pypi| |versions|

-  `Documentation`_

.. _Documentation: http://google-cloud-python-happybase.readthedocs.io/

This client is provided as a `HappyBase`_ emulation layer which uses
Google Cloud Bigtable as the underlying storage layer.

If you need support for other Google APIs, check out the
`Google APIs Python Client library`_.

.. _Google APIs Python Client library: https://github.com/google/google-api-python-client
.. _HappyBase: https://happybase.readthedocs.io/en/latest/

Quick Start
-----------

::

    $ pip install --upgrade google-cloud-happybase


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Python >= 3.4

Deprecated Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^
Python == 2.7. Python 2.7 support will be removed on January 1, 2020.


Authentication
--------------

With ``google-cloud-python`` we try to make authentication as painless as
possible. Check out the `Authentication section`_ in our documentation to
learn more. You may also find the `authentication document`_ shared by all
the ``google-cloud-*`` libraries to be helpful.

.. _Authentication section: http://google-cloud-python.readthedocs.io/en/latest/google-cloud-auth.html
.. _authentication document: https://github.com/googleapis/gcloud-common/tree/master/authentication

Contributing
------------

Contributions to this library are always welcome and highly encouraged.

See `CONTRIBUTING`_ for more information on how to get started.

.. _CONTRIBUTING: https://github.com/googleapis/google-cloud-python-happybase/blob/master/CONTRIBUTING.rst

License
-------

MIT - See `LICENSE`_ for more information.

.. _LICENSE: https://github.com/googleapis/google-cloud-python-happybase/blob/master/LICENSE

.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-happybase.svg
   :target: https://pypi.org/project/google-cloud-happybase/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-happybase.svg
   :target: https://pypi.org/project/google-cloud-happybase/


CHANGELOG.md
# Changelog

[PyPI History][1]

[1]: https://pypi.org/project/google-cloud-happybase/#history


## 0.33.0

03-18-2019 12:57 PDT


### New Features
- Enforce types more strictly on puts and handle non-encoded column names. ([#63](https://github.com/googleapis/google-cloud-python-happybase/pull/63))

### Internal / Testing Changes
- Remove badges ([#65](https://github.com/googleapis/google-cloud-python-happybase/pull/65))
- Fix tests ([#66](https://github.com/googleapis/google-cloud-python-happybase/pull/66))
- Add kokoro ([#61](https://github.com/googleapis/google-cloud-python-happybase/pull/61))
- Blacken ([#64](https://github.com/googleapis/google-cloud-python-happybase/pull/64))

## 0.32.1

01-23-2019 13:08 PST


### Implementation Changes
- Table.send: fix Python3 compatibility. ([#59](https://github.com/googleapis/google-cloud-python-happybase/pull/59))

## 0.32.0

12-17-2018 17:20 PST


### Implementation Changes
- Pool: fix call to `_get_instance`. ([#57](https://github.com/googleapis/google-cloud-python-happybase/pull/57))
- Use `Table.mutate_rows()` rather than `Row.commit()` in `Batch.send()`. ([#54](https://github.com/googleapis/google-cloud-python-happybase/pull/54))
- Remove `instance.copy()` and `timeout` parameter from `Connection` constructor. ([#50](https://github.com/googleapis/google-cloud-python-happybase/pull/50))

### Documentation
- Announce deprecation of Python 2.7 ([#55](https://github.com/googleapis/google-cloud-python-happybase/pull/55))

### Internal / Testing Changes
- Harden system tests. ([#53](https://github.com/googleapis/google-cloud-python-happybase/pull/53))
- Update github issue templates ([#52](https://github.com/googleapis/google-cloud-python-happybase/pull/52))
- Testing cleanups ([#49](https://github.com/googleapis/google-cloud-python-happybase/pull/49))
- Fix version in `setup.py`

## 0.31.0 (2018-10-05)

### Dependencies

- Update to use `google.cloud.bigtable >= 0.31.0`


## 0.30.2 (2018-10-05)

### Dependencies

- Update to use `google.cloud.bigtable 0.30.2`

### Documentation

- Fix incorrect PyPI link in README (#28).

### Implementation changes

- Use Bigtable's `yield_rows` for `Table.rows()` and `Table.scan()`. ([#37](https://github.com/googleapis/google-cloud-python-happybase/pull/37))
- Update scan method to use row set ([#42](https://github.com/googleapis/google-cloud-python-happybase/pull/42))
- Use `row_set` in `rows()` method.  Improves performance for the `rows` method. ([#44](https://github.com/googleapis/google-cloud-python-happybase/pull/44))


## 0.26.0 (2017-08-07)

### Dependencies

- Update to use `google.cloud.bigtable 0.26.0`

### Documentation

- Fix docs URL (#25).


## 0.25.0 (2017-07-19)

### Dependencies

- Update to use `google.cloud.bigtable 0.25.0`


## 0.24.0 (2017-03-31)

### New Features

- Add support for Python 3.6.

### Dependencies

- Update to use `google.cloud.bigtable 0.24.0`


## 0.23.0 (2017-02-24)

### Dependencies

- Update to use `google.cloud.bigtable 0.23.0`


## 0.22.0 (2016-12-10)

### Dependencies

- Update to use `google.cloud.bigtable 0.22.0`


## 0.21.0 (2016-11-22)

### Dependencies

- Update to use `google.cloud.bigtable 0.21.0`


## 0.20.0 (2016-09-29)

### New Features

- Initial release.


noxfile.py
import nox


@nox.session(python="3.6")
def cover(session):
    session.install("pytest", "mock", "coverage", "pytest-cov")
    session.install(".")
    session.run(
        "py.test",
        "--quiet",
        "--cov=google.cloud.happybase",
        "--cov=unit_tests",
        "--cov-config",
        ".coveragerc",
        "unit_tests",
    )


@nox.session(python="3.6")
def docs(session):
    session.install("pytest", "mock", "Sphinx", "sphinx_rtd_theme")
    session.install(".")
    session.run(
        "python",
        "-c",
        "import shutil; shutil.rmtree('docs/_build', ignore_errors=True)",
    )
    session.run(
        "sphinx-build",
        "-W",
        "-b",
        "html",
        "-d",
        "docs/_build/doctrees",
        "docs",
        "docs/_build/html",
    )
    session.run(
        "python", "scripts/verify_included_modules.py", "--build-root", "_build"
    )


@nox.session(python="3.7")
def lint(session):
    """Run linters.
    Returns a failure if the linters find linting errors or sufficiently
    serious code quality issues.
    """
    session.install("flake8", "black")
    session.run("black", "--check", "src", "docs", "unit_tests", "system_tests")
    session.run("flake8", "google", "tests")


@nox.session(python="3.6")
def blacken(session):
    session.install("black")
    session.run("black", "noxfile.py", "src", "docs", "unit_tests", "system_tests")


@nox.session(python=["2.7", "3.4", "3.5", "3.6", "3.7"])
def unit(session):
    session.install("pytest", "mock")
    session.install(".")
    session.run("py.test", "--quiet", "unit_tests")


@nox.session(python=["2.7", "3.7"])
def system(session):
    session.install("pytest", "mock")
    session.install("-e", ".")
    session.run("py.test", "--quiet", "system_tests/test_happybase.py")
